solidity/contracts/StakingPortBacker.sol

```diff
@@ -0,0 +1,273 @@
+
+    event StakeCopied(
+        address indexed owner,
+        address indexed operator,
+        uint256 value
+    );
+    event StakePaidBack(
+        address indexed owner,
+        address indexed operator
+    );
+    event TokensWithdrawn(uint256 amount);
+
+    /// @notice The maximum allowed time for the token owner to repay the
+    /// delegation backed by tokens from this contract. After this time, the
+    /// owner of the contract can undelegate and recover not paid back
+    /// delegations.
+    uint256 public constant maxAllowedBackingDuration = 7776000; // ~3 months
+
+    IERC20 public keepToken;
+    TokenGrant internal tokenGrant;
+    StakeDelegatable public oldStakingContract;
+    TokenStaking public newStakingContract;
+
+    struct CopiedStake {
+        address owner;
+        uint256 amount;
+        uint256 timestamp;
+        bool paidBack;
+    }
+
+    mapping(address => bool) public allowedOperators; // operator -> allowed to copy?
+    mapping(address => CopiedStake) public copiedStakes; // operator -> copied stake info
+
+    constructor(
+        IERC20 _keepToken,
+        TokenGrant _tokenGrant,
+        StakeDelegatable _oldStakingContract,
+        TokenStaking _newStakingContract
+    ) public {
+        keepToken = _keepToken;
+        tokenGrant = _tokenGrant;
+        oldStakingContract = _oldStakingContract;
+        newStakingContract = _newStakingContract;
+    }
+
+    /// @notice Lets the owner of the contract to register operator from staking
+    /// relaitonships that will be allowed to use the token supply provided by
+    /// this contract.
+    /// @dev The reason for explicitly listing which relationships are allowed
+    /// to use the supply is to avoid a situation when someone stakes on the old
+    /// contract later, after deploying the new staking contract version, just
+    /// to drain the supply from this contract.
+    /// @param operator The operator from the staking relationship on the old
+    /// staking contract that will be allowed to use the token supply provided
+    /// by this contract.
+    function allowOperator(address operator) public onlyOwner {
+        allowedOperators[operator] = true;
+    }
+
+    /// @notice Copies staking relationship from the old token staking contract
+    /// to the new staking contract. Only the owner of the relationship can copy
+    /// it to the new staking contract, the relationship can only be copied once,
+    /// and undelegation must not be complete on that relationship.
+    /// Operator, beneficiary, authorizer, and stake amount will be the same
+    /// as in the original delegation. Until the delegation is repaid, this
+    /// contract is the owner of the delegation on the new staking contract.
+    /// @param operator The operator from the staking relationship on the old
+    /// staking contract that should be copied to the new staking contract.
+    function copyStake(address operator) public {
```

Comment:
Do we expect the old stake to be undelegated before copying it to the new token staking contract?

Reply:
I had to think about it a little more but it appears to me it doesn't really matter. We are going to deploy a new operator contract and that new operator contract will refer only to the new staking contract. In the worst scenario, we can have someone staking on the old staking contract and using the old operator contract and staking on the new staking contract using the additional liquidity we provide and using the new operator contract. After tBTC is relaunched, it's going to use the new operator contract and even if we assume the old operator contract has the potential of providing some yields (unlikely), we can still force-undelegate from the new contract if the delegation is not paid back on time.

tl;dr; I don't think it's worth introducing additional complexity.